src/py/mat3ra/notebooks_utils/core/entity/job/api.py

```
@@ -0,0 +1,116 @@
+    Saves a file to disk, overwriting any files with the same name as filename_on_disk.
+
+    Args:
+        job_id (str): ID of the job
+        job_endpoint (JobEndpoints): Job endpoint object from the Exabyte API Client
+        filename_on_cloud (str): Name of the file on the server
+        filename_on_disk (str): Name the file will be saved to
+    """
+    files = job_endpoint.list_files(job_id)
+    file_metadata = next(f for f in files if filename_on_cloud in f["key"])
+    signed_url = file_metadata["signedUrl"]
+    server_response = urllib.request.urlopen(signed_url)
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)